Magento 2.4.1 Beta Release Notes (#7846)
magento/devdocs: src/guides/v2.4/release-notes/release-notes-2-4-1-commerce.md

## Other release information

Although code for these features is bundled with quarterly releases of the Magento core code, several of these projects (for example, B2B, Page Builder, and Progressive Web Applications (PWA) Studio) are also released independently. Bug fixes for these projects are documented in the separate, project-specific release information that is available in the documentation for each project.

## Highlights

Look for the following highlights in this release.

### Substantial security enhancements

This release includes over 15 security fixes and platform security improvements. All security fixes have been backported to Magento 2.4.0-p1 and Magento 2.3.6.

#### Over 15 security enhancements that help close remote code execution (RCE) and cross-site scripting (XSS) vulnerabilities

> No confirmed attacks related to these issues have occurred to date. However, certain vulnerabilities can potentially be exploited to access customer information or take over administrator sessions. Most of these issues require that an attacker first obtains access to the Admin. As a result, we remind you to take all necessary steps to protect your Admin, including but not limited to these efforts: IP allowlisting, [two-factor authentication]({{page.baseurl}}/security/two-factor-authentication.html), use of a VPN, the use of a unique location rather than `/admin`, and good password hygiene. See _Security Updates Available for Magento_ for a discussion of these fixed issues.

Author: yes, it's a placeholder -- link won't be available until GA

{:.bs-callout-info}
Starting with the release of Magento Commerce 2.3.2, Magento will assign and publish indexed Common Vulnerabilities and Exposures (CVE) numbers with each security bug reported to us by external parties. This allows users of Magento Commerce to more easily identify unaddressed vulnerabilities in their deployment. You can learn more about CVE identifiers at [CVE](https://cve.mitre.org/).

### Infrastructure improvements

This release contains enhancements to core quality, which improve the quality of the Framework and these functional areas: Customer Account, Catalog, CMS, OMS, Import/Export, Promotions and Targeting, Cart and Checkout, B2B, and Staging and Preview.

> *  **Site-Wide Analysis Tool (SWAT) integration with Magento Admin**. (SWAT) provides system insights and instrumentation for Commerce Cloud installations of Magento with 24/7 real-time performance monitoring, reports, and self-service recommendations. Merchants can use the new SWAT Admin role to securely access their SWAT Customer Detail pages through the Magento Admin. For more information, see [SWAT FAQ](https://support.magento.com/hc/en-us/articles/360048646671).<!--- SWAT-807-->

Author: I'll add this link (and others to devdoc topics) to the GA version of these notes

### Performance improvements

*  **Reduction in the size of network transfers between Redis and Magento**. Plugin list configuration is now generated during the execution of the `di:compile` command. This configuration information is written to generated metadata folders based on scope. Previously, this information was stored in cache. Resulting performance improvements include:

   *  Network cache size has decreased by 10 - 15%

>    *  Execution time for many scenarios has been improved by 3%.<!--- MC-31617-->

Author: changed in both Commerce and OS files

*  **Improved execution time** for `bin/magento` commands.

### Adobe Stock Integration

This release includes Adobe Stock Integration v2.1.0.

### New Media Gallery

The New Media Gallery is now enabled by default in the Admin. Merchants can now perform these actions on images in the Media Gallery:

*  Delete images in bulk
*  Optimize media storage by identifying duplicate images and images that are not used on the storefront
*  Filter images by the storefront area they are used in, including product and category content and CMS blocks
*  Work with image metadata

>    *  View meta data from the images uploaded into Media Gallery

Author: changed in both files

>    *  Edit image meta data (title, description, and keywords)

Author: Changed to metadata in both files

>    *  Search for images by their meta data

Author: Changed to metadata in both files

### Page Builder

Page Builder now supports full screen mode, which supports easier editing of content and provides a consistent experience editing content across the Admin. <!--- PB-543-->

### GraphQL

This release adds GraphQL coverage for the following features:

*  **Product reviews**. Customers and guests can write product reviews. Customers can retrieve their product review histories.<!--- MC-32349-->

> *  **Gift options**. All customers and guests add a gift message to their order. On EE installations, they can also add gift wrapping, gift receipts, and printed cards to the order.<!--- MC-32345-->

Author: Ooops! forgot about that EE

> *  **Reward points**. On EE installations, customers can apply or remove reward points to their carts. They can also view their reward point history.<!--- MC-23366-->

Author: got it -- thanks for deciphering my shorthand

*  **Order history**. All customers can view details about their order histories, including invoices, shipping, and refunds.<!--- MC-20635-->

> *  **Add to cart**. Customer can add the following product types to their cart:  Simple, Configurable, Bundled, Grouped, Virtual, Gift Card, and Downloadable. <!--- MC-21513-->

Author: Changed in both files

*  **Stored payment methods**. Logged-in customers can now store payment details (including Braintree credit card and Braintree with PayPal) in MyAccount. <!--- MC-32348 35945 35946-->

See the [GraphQL Developer Guide]({{page.baseurl}}/graphql/) for details on this and other enhancements.

### PWA Studio

PWA Studio v8.0.0 introduces new features and enhancements:

> *  updates to the Venia style guide that apply to design tokens, typography, colors, core components, and page layouts. <!--- PWA-519 419-->

Author: initial-capped each entry in this list

> *  improvements to the Venia mini-cart experience <!--- PWA-236-->

Author: initial-capped each entry in the list in both files

> *  initial support for multiple locales and localized content on the Venia storefront <!--- PWA-295-->

Author: initial-capped each entry in the list in both files

> *  numerous improvements to the MyAccount experience of the Venia storefront <!--- PWA-247-->

Author: initial-capped each entry in the list in both files

See [Magento compatibility](https://magento.github.io/pwa-studio/technologies/magento-compatibility/) for a list of PWA Studio versions and their compatible Magento core versions. For information about enhancements and bug fixes, see [PWA Studio releases](https://github.com/magento/pwa-studio/releases).

### B2B

Magento 2.4.1 introduces B2B v1.3.0. This release includes improvements to order approvals, shipping methods, shopping cart, and logging of Admin actions.

#### Improvements to Order Approvals

B2B order approvals have been enhanced to improve usability and to allow for bulk actions on purchase orders.
Improvements to order approval and rejection include the following:

> *  **New View Rule page for users without edit privilege**. B2B buyers can now view rules that apply to their company on the new View Rule page when they don’t have permission to edit them. <!--- BUNDLE-104 -->

Author: changed

*  **Count alert icon on the Requires My Approval tab**. The Requires My Approval tab in the My Purchase Orders view now displays a counter that indicates the number of pending approval actions. <!--- BUNDLE-102 -->

*  **Bulk order approvals and rejections**. B2B managers and Company Administrators can now perform bulk rejection and approval of purchase orders. These changes allow approvers to approve or reject multiple purchase orders in a single action. <!--- BUNDLE-140 128 111 153 -->

*  Merchants can now search the **Applies to** and **Requires approval from** fields and can select multiple user roles during rule creation. <!--- BUNDLE-105 106-->

> *  Examples of how to configure Order Approval rules <!--- BUNDLE-103 -->

Author: no, not a placeholder, but missing a few words.

#### B2B shipping methods enhancements

B2B merchants can now control shipping methods that are offered to each Company. Merchants can configure the following from the Admin:<!--- BUNDLE-160 161 162 -->

> *  a specific set of shipping methods for B2B Company accounts

Author: initial-capped all entries in this list

> *  the use of All or B2B-specific shipping methods for each Company account

Author: done

> *  a specific list of B2B shipping methods for each Company account

Author: done

#### Shopping cart improvements

*  Merchants can now allow users to clear the contents of their shopping cart in a single action and can configure this ability independently on each website <!--- BUNDLE-108 -->

*  B2B buyers can now add individual items or the entire contents of their shopping cart directly to a requisition list. <!--- BUNDLE-145 144-->

#### New Admin features

*  B2B merchants can create orders from the Admin on behalf of customers using Payment on Account as the payment method. <!--- BUNDLE-166 178-->

*  Merchants can now directly view all quotes associated with a user from the customer’s detail page. <!--- BUNDLE-139 -->
*  Merchants can now filter the Company Customers table by Company. <!--- BUNDLE-137 -->

*  Admins can now filter customers in the Admin by Sales Rep. <!--- BUNDLE-110 -->

#### Enhanced security on storefront

> To reduce creation of fraudulent or spam accounts, merchant can now enable Google reCAPTCHA on the New Company Request form on the storefront. <!--- BUNDLE-154 -->

Author: done

#### Expanded logging of Admin actions

Admin actions taken in the Company modules are now logged in the Admin Actions Log. Actions are logged from all relevant company modules: `Company`, `NegotiableQuote`, `CompanyCredit`, `SharedCatalog`.  <!--- BUNDLE-180 181 182 183 -->

This release also includes multiple bug fixes. See [B2B Release Notes]({{page.baseurl}}/release-notes/b2b-release-notes.html).

### Magento Functional Testing Framework (MFTF)

MFTF 3.1.0 is now available. See [Magento Functional Testing Framework Changelog](https://github.com/magento/magento2-functional-testing-framework/blob/develop/CHANGELOG.md).

## Fixed issues

We have fixed hundreds of issues in the Magento 2.4.1 core code.

### Installation, upgrade, deployment

<!--- MC-36113-->

> *  Installation of Magento with third-party extensions that have dependencies on APIs for the Store module in CLI commands no longer fails. Previously, Magento displayed this error message during: The default website isn't defined. Set the website and try again. This was a known issue in Magento 2.4.0.

Author: changed in both files

<!--- MC-17959-->

*  Running `/bin/magento config:show vendor_module/general/value` now returns `0` or an empty string as expected. Previously, it returned `Configuration for path: "vendor_module/general/value" doesn't exist`. [GitHub-23290](https://github.com/magento/magento2/issues/23290)

<!--- MC-33788-->

*  Upgrade no longer results in the sudden failure of the Galera cluster. Previously, the Galera cluster exited abruptly after re-indexing immediately after upgrade. During Magento upgrade, index tables are altered and the engine is changed from MEMORY to InnoDB. At this point, the content of these tables becomes out-of-sync between the nodes of the Galera cluster. [GitHub-25334](https://github.com/magento/magento2/issues/25334)

<!--- ENGCOM-7219-->

<!--- ENGCOM-7459-->

<!--- ENGCOM-7265-->

<!--- ENGCOM-7508-->

<!--- ENGCOM-7800-->

<!--- ENGCOM-7629-->

<!--- ENGCOM-7776-->

<!--- ENGCOM-7853-->

<!--- ENGCOM-7883-->

<!--- ENGCOM-8006-->

<!--- ENGCOM-8020-->

<!--- ENGCOM-7987-->

### AdminGWS

<!--- MC-36164-->

> *  Magento no longer displays the **Add Attribute** button (**Stores** > **Attributes** ) or **Add Attribute Set** button (**Stores** > **Attributes** > **Customer** ) buttons when the logged-in administrator lacks the appropriate permissions to create these entities. Previously, Magento threw a 404 error when a website administrator who did not have the appropriate permissions tried to create an **Attribute Set** or **Customer** attribute.

Author: removed redundant button

<!--- MC-36330-->

<!--- MC-36230-->

*  Magento no longer throws an error when an administrator with restricted roles for specific websites tries to create a subcategory from the Admin.

### Analytics

<!--- MC-33314-->

<!--- MC-34352-->

> *  Magento successfully generates advance reporting data files and sends them as expected to Inventory on deployments with split databases. Previously, Magento didn’t generate or send the `quotes.csv` file to Inventory, and as a result, Inventory did not generate the expected reports.

Author: changed in Commerce file

### Bundle products

<!--- MC-36281-->

*  Magento no longer throws an exception when you try to create a product in a deployment in which MSI is installed but the `Magento_InventoryBundleProduct` module is disabled.

<!--- MC-34261-->

*  Magento now correctly calculates offline refunds for orders that contain bundle products.

<!--- MC-29908-->

<!--- ENGCOM-7499-->

<!--- ENGCOM-7655-->

<!--- ENGCOM-7985-->

### Cache

<!--- MC-36096-->

*  Local cache storage is now retained for the period of time set in **Stores** > **Configuration** > **General** > **Web** > **Default Cookie Settings**. Previously, the expiry date of cookies was hardcoded to one day, which put it out of sync with this setting. As a result, welcome messages did not retain returning customer information for the expected duration.

<!--- ENGCOM-7780-->

<!--- ENGCOM-7073-->

<!--- ENGCOM-8019-->

### Cart and checkout

<!--- MC-36252-->

<!--- MC-35329-->

*  The **Delete** button on the **Add to Shopping Cart by SKU** section of a customer’s **Manage Shopping Cart** page now works as expected when multiple rows are selected.

<!--- MC-34999-->

> *  Magento no longer throws an error when you try to order a product by SKU when the digits you enter match a valid SKU but the case of these digits differ. Previously, when you entered an SKU on My **Account** > **Order by SKU** that did not exactly match a valid SKU, Magento threw an error.

Author: keeping this as is  under the assumption that SKU is pronounced "S-K-U", now "skew". If the latter, then it needs to change to "a"

<!--- MC-25042-->

> *  A customer’s shipping address is now selected by default at checkout when the address is located in the country identified on the Allow Countries list and that list includes only that country. Previously, Magento didn’t select the address as default and displayed this error message: `Please specify a regionId in shipping address`.

Author: reverted all contractions throughout both files

<!--- MC-24043-->

*  Merchants can now enable **Apply to Shipping Amount** in the Action tab of **Marketing** > **Cart Price Rules** > **Add New Rule** when **Fixed amount discount for whole cart** is applied. [GitHub-24422](https://github.com/magento/magento2/issues/24422)

<!--- MC-23992-->

<!--- MC-33899-->

> *  Magento now displays an add-to-cart success message when customer adds an out-of-stock product to their cart. Previously, the product was added, but Magento didn’t display a success message.

Author: changed in both files

<!--- MC-35989-->

> *  Custom address attributes are now included as expected in the the form displayed for Payment step of checkout workflow.

Author: entered in both files. also deleted the redundant "the"

<!--- MC-36060-->

*  The **State/Province/Region** input box is now enabled as expected on **My account** > **Address Book** > **Add new address**.

<!--- ENGCOM-7746-->

<!--- ENGCOM-7752-->

<!--- ENGCOM-7585-->

<!--- ENGCOM-7457-->

<!--- ENGCOM-5629-->

<!--- ENGCOM-7949-->

<!--- ENGCOM-7968-->

<!--- ENGCOM-7976-->

<!--- ENGCOM-7976-->

<!--- ENGCOM-7976-->

<!--- ENGCOM-7976-->

<!--- ENGCOM-7976-->

<!--- ENGCOM-8019-->

<!--- ENGCOM-7825-->

<!--- ENGCOM-8004-->

<!--- ENGCOM-7950-->

<!--- MC-36418-->

*  Magento now displays a customer registration form when a guest user completes checkout.

<!--- MC-35607-->

### Catalog

<!--- MC-25062-->

<!--- ENGCOM-7292-->

<!--- ENGCOM-7563-->

<!--- ENGCOM-7513-->

<!--- ENGCOM-7420-->

<!--- ENGCOM-7976-->

<!--- MC-34314-->

> *  Magento now uses the default option (**Configuration** > **Web** > **Default Layouts** > **Default Product Layout**) that you’ve selected for the `page_layout` attribute when creating a new product . Previously, your selected default value wasn’t applied.

Author: done

<!--- MC-34258-->

> *  You can now successfully perform mass actions on Inventory product stock. Previously, when you tried to perform a mass action inventory product stock, Magento displayed a blank page. If you performed this action with Magento in developer mode, Magento threw this error: `Notice: Undefined offset: 32000 in /Users/kodithuw/sites/m23inventory/inventory/InventoryCatalogAdminUi/view/adminhtml/templates/catalog/product/edit/action/inventory.phtml on line 24`

Author: added to both files

<!--- MC-32552-->

*  The total record count displayed when you click **Add Products** on the Products tab when adding products to a catalog category no longer changes based on product sort order.

### CMS content

<!--- MC-35971-->

*  The Hierarchy tab for a selected default store view now displays the selected parent page as expected.

<!--- ENGCOM-7600-->

<!--- ENGCOM-7602-->

<!--- MC-35480-->

### Configurable products

<!--- ENGCOM-7214-->
<!--- ENGCOM-7787-->
<!--- MC-33523-->

*  Pagination problems with the Configurable Product Edit Current Variations list have been corrected.

<!--- MC-33406-->

<!--- MC-29882-->

*  Magento now displays the correct price for configurable products with customizable options on the Admin Create Order page. As expected, the displayed price is a sum of the child product's price and the custom option's price. Previously, Magento displayed only the custom option price. [GitHub-25766](https://github.com/magento/magento2/issues/25766)

<!--- MC-33745-->

> *  Order summary now displays the correct discount amount when a cart price rule has been applied. Previously, the rule didn’t correctly round amounts when calculated shipping discounts.

Author: changed in both files

<!--- MC-33765-->

> *  Admin users accounts created from an admin account with a restricted scope can now create a configurable product with attributes as expected. Previously, Magento threw this error: `Notice: Undefined index: value_index in 23develop/app/code/Magento/ConfigurableProduct/Helper/Product/Options/Factory.php on line 101`.

Author: entered in both files

### Custom customer attributes

<!--- MC-36387-->

*  Corrected alignment issues for the explanatory text about passwords and the **Job Title** field on the Customer Edit Account Information page.

<!--- MC-33645-->

<!--- MC-34024-->

*  The checkout workflow no longer displays custom customer address attribute values when the customer has not entered any data.

### Customer

<!--- MC-33679-->

*  The region names in Admin customer addresses are now translated as expected.

<!--- ENGCOM-7793-->
<!--- MC-36226-->

*  The **State/Province** fields are now populated as expected on the Edit Address page (**My Account** > **Address book**).

<!--- MC-34655-->

*  Magento no longer throws an error when a customer clicks the **Submit** button multiple times on forms throughout the storefront for which invisible reCAPTCHA has been enabled. Previously, clicking this button multiple times resulted in an internal error similar to this: `Internal error: Make sure you are using reCaptcha V3 api keys`.

<!--- MC-33522-->
<!--- MC-33357-->
<!--- MC-33150-->
<!--- MC-35655-->

> *  Magento no longer displays the **Delete customer** button on the **Customers** page when the logged-in administrator does not have rights to delete customers in deployments where B2B is installed.

Author: done

### Customer segment

<!--- MC-33184-->
<!--- MC-36224-->

*  Customer segment conditions now work as expected in a split-database deployment. Previously, Magento threw an error when you tried to edit a customer segment by adding a condition: `SQLSTATE[42S02]: Base table or view not found`

### Directory

<!--- MC-32271-->

*  The format of the State/Province drop-down menu is now consistent across the Admin.

### Downloadable

<!--- MC-29905-->

<!--- MC-35026-->

*  The My Downloadable Products area now displays links to purchased downloadable products that are part of a grouped product as expected.

<!--- MC-34262-->

> *  Clicking on a downloadable’s product **Sample** button from the Admin product page now downloads a sample as expected. Previously, when you clicked **Sample**, Magento displayed this error: `The product that was requested doesn't exist. Verify the product and try again`.

Author: entered in both files

<!--- ENGCOM-7757-->

<!--- ENGCOM-7796-->

### Dynamic block (formerly banner)

<!--- MC-33266-->

> *  Table title now matches the data table.as expected when you create a dynamic block and add a related catalog price rule.

Author: changed in Commerce

<!--- MC-33286-->

### Email

<!--- MC-32789-->

> *  Magento now sends email notifications about order changes to the correct customer email if the customer email has been changed after the order was created.

Author: changed in both files

<!--- MC-36015-->

*  Order update emails that are sent to customers now include the correct order status. Previously, if an order status changed from `processing` to another state, the order email did not reflect the status change.

<!--- MC-34107-->

*  Magento no longer displays misleading messages about existing accounts during guest checkout. Previously, when a guest navigated to the checkout page, then navigated back to the shipping page, Magento displayed this error: `You already have an account with us. Sign in or continue as guest`.

<!--- MC-33905-->

> *  Custom email templates now load the same elements as native default email templates. Previously, some elements, including  variable values, were missing.

Author: removed tiny extraneous space in both files

<!--- MC-33700-->

<!--- ENGCOM-7576-->

<!--- ENGCOM-7177-->

<!--- ENGCOM-7506-->

<!--- ENGCOM-7815-->

<!--- MC-33232-->

### Frameworks

<!--- MC-35893-->

*  The **Time of day to send data** field of the Admin **Stores** > **Configuration** > GENERAL > **Advanced Reporting** page is now rendered correctly.

<!--- MC-34153-->

> *  Shoppers can now change the number of orders that are displayed per page when the Orders list spans multiple pages. Previously, Magento displays this message when you navigate to the last page of orders and try to change the number of orders displayed per page: `You have placed no orders`. This was a known issue for Magento 2.4.0.

Author: done

<!--- MC-35020-->

*  You can now add products to a category when implementing Level 2 cache.

<!--- MC-29755-->

> *  `X-Magento-Tags` headers no longer exceed the size permitted by the HTTP specification. Previously, category pages that contain  many products return an `X-Magento-Tag` header that resulted in a 503 error.

Author: changed in both files

<!--- MC-34257-->

*  `sales_order_shipment_track_save_commit_after` is now triggered as expected when you use REST to create a shipment.

<!--- MC-33898-->

*  Magento now displays an informative error message when a `di compile` fails due to a nonexistent dependency. Previously, the message displayed did not identify the class in which the exception occurred.

<!--- MC-32929-->

*  Magento no longer throws the following fatal error when Redis uses all allowed memory: `report.CRITICAL: OOM command not allowed when used memory > 'maxmemory'.`

### General fixes

<!--- MC-36048-->

*  Sorting products on the Admin (**Stores** > **Attributes** > **Products**) now displays all products that meet your search criteria. Previously, Magento did not display records, and to sort the records, you needed to navigate to the first page of search results.

<!--- MC-36343-->

*  Unnecessary quotation marks and escaping around a URL in `tracking.phtml` have been removed.

<!--- MC-35998-->

> *  The `var/log/system.log` now displays a more accurate message when a user tries to access non-existing resource file under the static directory when SCD OnDemand and production mode are enabled. Previously, Magento logged the same message that is logged when the error occurs in developer mode. Magenta now logs a 404 error.

Author: done

<!--- ENGCOM-7286-->

<!--- ENGCOM-7532-->

<!--- ENGCOM-7577-->

<!--- ENGCOM-7591-->

<!--- ENGCOM-7447-->

<!--- ENGCOM-7299-->

<!--- ENGCOM-7193-->

<!--- ENGCOM-7610-->

<!--- ENGCOM-7565-->

<!--- ENGCOM-7682-->

<!--- ENGCOM-7514-->

<!--- ENGCOM-7588-->

<!--- MC-33744-->

<!--- ENGCOM-7511-->

<!--- ENGCOM-7462-->

<!--- ENGCOM-7729-->

<!--- ENGCOM-7810-->

<!--- ENGCOM-7801-->

<!--- ENGCOM-7816-->

<!--- ENGCOM-7724-->

<!--- ENGCOM-7844-->

<!--- ENGCOM-7834-->

<!--- ENGCOM-7854-->

<!--- ENGCOM-7826-->

<!--- ENGCOM-7147-->

<!--- ENGCOM-7868-->

<!--- ENGCOM-7867-->

<!--- ENGCOM-7881-->

<!--- ENGCOM-7857-->

<!--- ENGCOM-7856-->

<!--- ENGCOM-7999-->

<!--- ENGCOM-7962-->

<!--- MC-35550-->

<!--- MC-35230-->

> *  Password lifetime as set in **Stores** > **Configuration** > **Advanced** > **Admin** is now honored. Previously, If you clicked **Forgot password?** when prompted to reset your password, you could bypass the password reset.

Author: done. great eyes!

<!--- MC-34369-->

*  Coupon codes are now applied only to the specified product. Previously, Magento applied the coupon code to all products in the cart. [GitHub-28246](https://github.com/magento/magento2/issues/28246)

### Gift cards

<!--- ENGCOM-7661-->

<!--- MC-36118-->

*  Gift Card Accounts now capture order numbers as expected. Previously, the **More information** field in the History tab for the selected gift account did not display order IDs.

<!--- MC-34519-->

*  Using a comma as a decimal separator now works as expected. Previously, the comma separator ignored decimal values.

<!--- MC-34469-->

> *  Gift cards are now displayed as expected on the mini cart. Previously, Magento rendered HTML objects as text.

Author: entered in Commerce only

<!--- MC-33851-->

*  Credit memos now correctly reflect the grand total for orders that involved discounted products and that were paid for by a combination of git card and store credit.

### Google Tag Manager

<!--- MC-33729-->

*  Magento no longer throws a JavaScript error during checkout when the **Cookie Restriction Mode** setting and Google Tag Manager are enabled.

### Import/export

<!--- MC-35479-->

> *  The `error_report.csv` file now downloads with content and is available inside the `var/import_history/` directory as expected. Previously, this file wasn’t generated after import.

Author: done

<!--- ENGCOM-7616-->

<!--- ENGCOM-7673-->

<!--- ENGCOM-7995-->

<!--- MC-33730-->

> *  Magento now successfully imports customer addresses that contains a region for a country that does not have defined regions. Previously, Magento threw this error: `Please enter a valid region`.

Author: done

<!--- MC-33277-->

*  Magento now loads the correct Entity Attributes set when a merchant selects an Entity Type when scheduling a new export.

<!--- MC-32956-->

> *  The position of products in the `catalog_category_product` table now updates as expected when an administrator creates a product in the Admin and assigns it to a category. Previously, the position of new products were always assigned a 0 value.

Author: done

<!--- MC-34939-->

*  Customer data is now successfully exported from the Admin, and the export data grid displays customer data as expected. Previously, an error related to memory allocation occurred during export.

<!--- MC-34657-->

> *  Imported `.cvs` files now capture related product information as expected. Previously, related product information was not consistently uploaded the first time the .cvs file was imported.

Author: added formatting

<!--- ENGCOM-7522-->

<!--- ENGCOM-7788-->

<!--- ENGCOM-7739-->

### Payment methods

<!--- MC-34363-->

*  Magento now displays a message that prompts you to enter mandatory credit card data when you click **Submit** for an Admin order without entering valid payment information. Previously, the Braintree card validator did not throw an error when payment input fields were invalid, and the page became inactive.

<!--- MC-33494-->

*  You can now change the shipping method for an order you create from the Admin for a customer whose account has a stored credit card (Braintree). Previously, when you selected a different shipping method, the stored card was not selected, and Magento didn’t place the order.

#### PayPal

<!--- MC-33879-->

*  The Order Review page of the checkout workflow now displays the correct shipping amount for PayPal through Braintree orders for which the shipping method has been changed during checkout. Previously, when a customer changed the shipping method on the PayPal Order Review page of the checkout workflow, Magento did not update the order total with the correct method.

<!--- MC-34152-->

> *  Merchants can now successfully cancel orders that were authorized using Merchant is unable to cancel an older order that was authorized using PayPal. Previously, Magento didn’t cancel the order and displayed this error:  `Declined: 10601-Authorization has expired`.

Author: copy-paste nastiness! thanks for catching it -- will change in JIRA too

<!--- MC-33330-->

*  Magento no longer empties your cart when you cancel an order by closing the PayPal payment popup window after first completing another order.

### Performance

<!--- MC-31617-->

> *  Plugin list configuration is now generated during the execution of the `di:compile` command. This configuration information is written to generated metadata folders based on scope. Previously, this information was stored in cache. Resulting performance improvements include:

Author: yes, this was intentional. we duplicate highlights where applicable in the fixed issue section

<!--- MC-35370-->

*  Corrected unnecessary scrolling on the Create New Order page.

<!--- MC-35345-->

*  Directly clicking on the **Export Tax Rates** button of the Add New Tax Rule page ( **Stores**  >  **Tax Rules**) now downloads the `tax_rates.csv` file as expected. Previously, merchants had to click on the edge of the **Export Tax Rates** button. This was a known issue in Magento 2.4.0.

<!--- MC-35313-->

*  The **Add selections to my cart** button on the bottom of the shopping cart now works as expected. This was a known issue in Magento 2.4.0.

<!--- MC-35296-->

*  The **Refresh** button of the **Recently Viewed Products** section of the Customer's Activities page now works as expected. Previously, when you clicked **Refresh** , the product list wasn’t refreshed, and the page scrolled.

<!--- MC-35412-->

*  Magento now correctly displays the calendar icon used for selecting a customer’s date of birth on the Conditions tab of **Customers** > **Segments** > **Add Segment**.

<!--- MC-35658-->

*  The checkout summary section of the checkout workflow no longer flickers when the customer scrolls through this page on Internet Explorer 11.x.

<!--- MC-34602-->

> *  Magento now correctly displays the Order by SKU widget on the storefront Category page. Previously, the HTML code for this widget wasn’t rendered, and Magento didn’t display the **Load a list of SKUs** link.

Reviewer: ```suggestion
*  Magento now correctly displays the Order by SKU widget on the storefront Category page. Previously, the HTML code for this widget was not rendered, and Magento did not display the **Load a list of SKUs** link.
```

<!--- ENGCOM-7619-->

<!--- ENGCOM-7873-->

<!--- ENGCOM-7639-->

<!--- ENGCOM-7156-->

<!--- ENGCOM-7981-->

<!--- ENGCOM-7911-->

<!--- ENGCOM-8007-->

<!--- ENGCOM-7992-->

<!--- ENGCOM-7967-->

<!--- ENGCOM-8015-->

### URL rewrites

<!--- MC-34483-->

*  You can now successfully preview the staging schedule for a CMS block. Previously, Magento threw a fatal error.

<!--- MC-33028-->

*  Magento now preserves existing catalog URL rewrites as expected when a store view is assigned to a different store. Previously, Magento deleted the store-specific URL rewrites.

### Vault

<!--- MC-34674-->

*  Magento no longer saves credit card numbers when the **Save for later use** checkbox on the payment section of the checkout workflow is not selected.

### Visual Merchandiser

<!--- MC-35574-->

*  Magento now displays source stock instead of the default product stock when you sort products in Visual Merchandiser and Inventory is enabled.

### Web API framework

<!--- ENGCOM-7618-->

<!--- ENGCOM-7709-->

<!--- ENGCOM-7665-->

<!--- ENGCOM-7612-->

<!--- ENGCOM-7785-->

<!--- ENGCOM-7715-->

<!--- ENGCOM-7829-->

<!--- ENGCOM-7611-->

<!--- MC-36419-->

<!--- MC-36084-->

> *  Invoices created using REST now include gift card information similar to how as invoices created in the Admin do. Previously, using POST `\{host}/rest/default/V1/order/3/invoice` to invoice the order didn’t display the gift card code or gift card amount applied.

Reviewer: ```suggestion
*  Invoices created using the REST API now include gift card information similar to invoices that were created using the Admin. Previously, using POST `\{host}/rest/default/V1/order/3/invoice` to invoice the order did not display the gift card code or gift card amount applied.
```

<!--- MC-35838-->

> *  Merchants with multiple websites can now use REST to create and update products while preserving  image and image-role inheritance. Previously, when a merchant used REST to create and update products, and a product was updated for store view, the default image roles were loaded and saved for that store view. As a result, the store-view image roles stopped inheriting from the default scope after update.

Reviewer: ```suggestion
*  Merchants with multiple websites can now use the REST API to create and update products while preserving  image and image-role inheritance. Previously, when a merchant used the REST API to create and update products, and a product was updated for store view, the default image roles were loaded and saved for that store view. As a result, the store-view image roles stopped inheriting from the default scope after update.
```

<!--- MC-36419-->

> *  An unscoped integration user  account can now access a resource through REST when resource permissions allow access.

Reviewer: ```suggestion
*  An unscoped integration user account can now access a resource through the REST API when resource permissions allow access.
```

Author: changed in both Commerce and OS files

<!--- MC-35975-->

### Website Restrictions

<!--- MC-32366-->

*  Enabling Website Restriction no longer blocks Varnish ESI requests for customers logged in as guest. Previously, these guests could not access the home page main menu.

<!--- MC-32369-->

*  Magento now loads the product home page as expected when website restrictions are enabled.

### Wishlist

<!--- MC-35810-->

> *  You can now use the wishlist search feature to find a product in a public wish list in deployments where support for multiple wishlists is enabled. Previously, after a customer used the wishlist search to find a product, selecting it, and clicking **Add to cart**, Magento didn’t add the product to the cart and displayed this error: `Invalid Form Key. Please refresh the page`.

Reviewer: Is "wishlist" one or two words? I see both variations in this paragraph.

Author: will change to "wish list" in both files. That's our glossary spelling, although industry standard is "wishlist'.

> *  You can now use the wishlist search feature to find a product in a public wish list in deployments where support for multiple wishlists is enabled. Previously, after a customer used the wishlist search to find a product, selecting it, and clicking **Add to cart**, Magento didn’t add the product to the cart and displayed this error: `Invalid Form Key. Please refresh the page`.

Reviewer: ```suggestion
*  You can now use the wishlist search feature to find a product in a public wish list in deployments where support for multiple wishlists is enabled. Previously, after a customer used the wishlist search to find a product, selecting it, and clicking **Add to cart**, Magento did not add the product to the cart and displayed this error: `Invalid Form Key. Please refresh the page`.
```

Author: Changed globally in both files.

<!--- MC-35622-->

> *  Customers can now change the quantity of a product in a wishlist from the wishlist itself. Previously, Magento did not update the product quantity and didn’t display a message.

Reviewer: ```suggestion
*  Customers can now change the quantity of a product in a wishlist from the wishlist itself. Previously, Magento did not update the product quantity and did not display a message.
```

Author: Changed globally in both files

<!--- MC-35618-->

> *  Customers can now change the quantity of a grouped product in a wishlist from the wishlist itself. Previously, Magento didn’t update the product quantity or display a message.

Reviewer: ```suggestion
*  Customers can now change the quantity of a grouped product in a wishlist from the wishlist itself. Previously, Magento did not update the product quantity or display a message.
```
